flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskTest.java

```diff
@@ -207,9 +209,8 @@ public void testCleanUpExceptionSuppressing() throws Exception {
 			testHarness.waitForTaskCompletion();
 		}
 		catch (Exception ex) {
-			if (!ExceptionUtils.findThrowable(ex, ExpectedTestException.class).isPresent()) {
-				throw ex;
-			}
+			assertTrue(ex.getCause() instanceof ExpectedTestException);
+			assertTrue(Iterables.getOnlyElement(asList(ex.getCause().getSuppressed())) instanceof FailingTwiceOperator.DisposeException);
```

Comment:
The previous construct that was checking if the exception as as expected and if not, rethrowing this unexpected exception was by design and intentional - if test fails because of some unexpected exception, it greatly speeds up figuring out what has failed, especially if that's a flaky failure.

Could this functionality be preserved? Replace `assertTrue(condition)` with `if (!condition) { throw ex; }`?

Sorry for nit picking on this and adding one more review round, but we (me personally included) wasted quite a bit of time because of this issue in various tests :(

Reply:
I think you are talking about tests that assert exception types unnecessarily.
But here we indeed want to make sure that `DisposeException` **is** in the list of suppressed.

Reply:
Yes, right. But in case of some bug, like NPE, or some other unexpected failure, it would be better to know immediately from the exception (visible in Azure) to know what has happened. Instead of just getting `expected true but was false` message.

Slightly better but IMO an overkill version would be to:
1. check if top level exception is `ExpectedTestException`, if not, throw the original exception
2. check if there is exactly one single expected suppressed exception. if not:
3. If there is more then one, or just one but different then expected, throw original
4. if there is no suppressed exception, throw `Assert.fail("Missing suppressed DisposeException");`

My proposal for double
```
if (!condition1) { throw ex; }
if (!condition2) { throw ex; }
```
Is just simpler and good enough, as it presents all of the required information to debug the test without re-running it in the debugger/with extra debug messages.